MatrixSDK/Data/MXRoomLastMessage.m

```diff
@@ -87,11 +101,29 @@ - (instancetype)initWithManagedObject:(MXRoomLastMessageMO *)model
         _originServerTs = model.s_originServerTs;
         _isEncrypted = model.s_isEncrypted;
         _sender = model.s_sender;
-        _text = model.s_text;
-        if (model.s_attributedText)
+        
+        NSData* attributedTextData;
+        if (model.s_isEncrypted)
         {
-            _attributedText = [NSKeyedUnarchiver unarchiveObjectWithData:model.s_attributedText];
+            attributedTextData = [self decrypt:model.s_attributedText];
         }
+        else
+        {
+            attributedTextData = model.s_attributedText;
+        }
+        
+        if (attributedTextData)
+        {
+            _attributedText = [NSKeyedUnarchiver unarchiveObjectWithData:attributedTextData];
+        }
+        
+        if (_attributedText == nil && model.s_text)
+        {
+            _attributedText = [[NSAttributedString alloc] initWithString:model.s_text];
```

Comment:
Does this not change the existing behaviour? Previously if `_attributedText` was unset, it would not be created from `model.s_text`.  It is not obvious to me whether `text` and `attributedText` are always variants of the same data or not. And if we always end up using `attributedText` whether it would be simpler to remove `text` entirely